infrastructure/testdriver/set_permission.html

```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>TestDriver set_permission method</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+
+<script>
+async_test(t => {
+  let permission_params = {
```

Comment:
This object is fairly verbose, will test authors need to change each part of it? Could the API be just `set_permission("ambient-light-sensor", "granted")`, at least a convenience variant?

Looking at tests which would use this API would be a good way to settle this question if there's doubt.

Reply:
Yeah, I considered this but thought sticking to the spec was important here. But agreed that just passing the name and state is much nicer for test authors, so done.
Note that I only simplified the `window.test_driver` class, while still having `test_driver_internal` take the dict as specced - let me know if that's a problem.

Reply:
Having `test_driver_internal` map quite directly to the WebDriver API sounds good!